File: Beecrowd/1049/1049.cpp
#include<iostream>

using namespace std;

int main(){

  string pri, sec, ter;

  cin >> pri >> sec >> ter;

  if(pri == "vertebrado"){
    if(sec == "ave"){
      if(ter == "carnivoro")
        cout << "aguia" << endl;
      else
        cout << "pomba" << endl;
    }
    else{
      if(ter == "onivoro")
        cout << "homem" << endl;
      else
        cout << "vaca" << endl;
    }
  }
  else{
    if(sec == "inseto"){
      if(ter == "hematofago")
        cout << "pulga" << endl;
      else
        cout << "lagarta" << endl;
    }
    else{
      if(ter == "hematofago")
        cout << "sanguessuga" << endl;
      else
        cout << "minhoca" << endl;
    }
  }

  return 0;
}

File: Beecrowd/1049/1049.py
# Recebe string de entrada
tipo1 = input()
tipo2 = input()
tipo3 = input()

# Vertebrado
if tipo1 == 'vertebrado':

  # Ave
  if tipo2 == 'ave':

    # Carnívoro
    if tipo3 == 'carnivoro':

      # Aguia
      print("aguia\n")
    else:

      #Pomba
      print('pomba\n')

  # Mamifero
  else:

    # Carnívoro
    if tipo3 == 'carnivoro':

      # homem
      print("homem\n")
    else:

      # vaca
      print('vaca\n')

# Invertebrado
else:

  # inseto
  if tipo2 == 'inseto':

    # hematofago
    if tipo3 == 'hematofago':

      # pulga
      print("pulga\n")
    else:

      #lagarta
      print("lagarta\n")

  # anelideo
  else:

    # hematofago
    if tipo3 == 'hematofago':

      # pulga
      print("sanguessuga\n")
    else:

      #lagarta
      print("minhoca\n")
